libs/jsonschema/utils.go

```diff
@@ -111,3 +112,32 @@ func FromString(s string, T Type) (any, error) {
 	}
 	return v, err
 }
+
+func ValidatePatternMatch(name string, value any, propertySchema *Schema) error {
+	if propertySchema.Pattern == "" {
+		// Return early if no pattern is specified
+		return nil
+	}
+
+	// Expect type of value to be a string
+	stringValue, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("invalid value for %s: %v. Expected a value of type string", name, value)
+	}
+
+	match, err := regexp.MatchString(propertySchema.Pattern, stringValue)
+	if err != nil {
+		return err
+	}
+	if match {
+		// successful match
+		return nil
+	}
+
+	// If custom user error message is defined, return error with the custom message
+	msg := propertySchema.PatternMatchFailMessage
+	if msg == "" {
+		msg = fmt.Sprintf("Expected to match regex pattern: %s", propertySchema.Pattern)
+	}
+	return fmt.Errorf("invalid value for %s: %s. %s", name, value, msg)
```

Comment:
Should we still stick to the Go convention of lowercasing these errors? :/ cc @pietern 

Assuming we should, then we should change databricks_template_schema.yml's failure message to start with a lower-case letter as well.

Reply:
Isn't the following OK? (with `<>` to indicate the verbatim failure message)

```
Error: invalid value for bar: axyze. <My custom message.>
```

@shreyas-goenka Can you also put quotes around the value? Otherwise empty string shows up as nothing. You can use `%q` instead of `%s` to quote it.

Reply:
Added quotes to the string value returned in the error message.

Reply:
@lennartkats-db Go style guide does not seem to have anything against capitalization in words other than the start of the error. The linter also does not complain for errors like `fmt.Errorf("foo. Bar")`. 

relevant style guide bits: https://google.github.io/styleguide/go/decisions.html#error-strings